thunder/core/baseutils.py

```diff
@@ -442,10 +442,10 @@ def print_base_printable(x: Any, /) -> str:
 _exec_ctr = 0
 
 
-def compile_and_exec(fn_name: str, python_str: str, program_name: str, ctx: dict) -> Callable:
+def build_callable(fn_name: str, python_str: str, file_name: str, ctx: dict) -> Callable:
```

Comment:
I must admit I'm puzzled.

Reply:
Yeah. Basically, `ctx` gets repurposed as `globals()` by `exec()`. Then we pull the function object from that globals dict. Don't know why we decided to do it this way, but it works.

Reply:
OK.